### PR TITLE
Xiaomi Redmi Note 10 Pro (sweet) M2101K6G config

### PR DESCRIPTION
**Pull request by BjornW**
I've added the config for the Xiaomi Redmi Note 10 Pro (sweet) with modelnr: M2101K6G (with some help from @anon1892, thank you!) . According to the [LineageOS wiki](https://wiki.lineageos.org/devices/sweet/variant1/) there are three variants of this phone. This config has been created and successfully tested for the M2101K6G model. 

I've also added this phone to the table of supported devices in the README.md. 

This pull-request would solve my own requested support issue #404 

Please let me know if there are any issues with this PR preventing it from being merged. Thanks!

### CODE REVIEW

File: openandroidinstaller/assets/configs/sweet.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+metadata:
+  maintainer: BjornW
+  brand: xiaomi
+  device_name: Xiaomi Redmi Note 10 Pro
```
Comment (MagicLike) [inline openandroidinstaller/assets/configs/sweet.yaml:4]:
```suggestion
  device_name: Xiaomi Redmi Note 10 Pro
  brand: xiaomi
```
Just a line order change as documented [here](https://github.com/openandroidinstaller-dev/openandroidinstaller/blob/main/docs/how_to_contribute_your_own_installation_configurations.md#how-to-write-metadata)

File: openandroidinstaller/assets/configs/sweet.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  is_ab_device: false
+  device_code: sweet
+  supported_device_codes:
+    - sweet
+    - sweetin
+  untested: false
+  notes:
+    - You should install Android 10 or newer ROM.
```
Comment (MagicLike) [inline openandroidinstaller/assets/configs/sweet.yaml:12]:
```suggestion
  notes:
    - You should install Android 10 or newer ROM.
```

`untested: false` is not needed, as it is set by default.

File: openandroidinstaller/assets/configs/sweet.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+metadata:
+  maintainer: BjornW
+  brand: xiaomi
+  device_name: Xiaomi Redmi Note 10 Pro
+  is_ab_device: false
+  device_code: sweet
+  supported_device_codes:
+    - sweet
+    - sweetin
+  untested: false
```
Comment (MagicLike) [inline openandroidinstaller/assets/configs/sweet.yaml:10]: Maybe also link the TWRP page here, with `twrp-link`, see [here](https://github.com/openandroidinstaller-dev/openandroidinstaller/blob/main/docs/how_to_contribute_your_own_installation_configurations.md#how-to-write-metadata)

**Review by MagicLike — CHANGES_REQUESTED** [review]

File: openandroidinstaller/assets/configs/sweet.yaml (hunk 3 above)
Comment (tsterbak) [reply openandroidinstaller/assets/configs/sweet.yaml:10]: Good point :) This is not strictly required, but if there is a custom link necessary and the link is not just resolved by knowing the device code, then having the link is nice.

**Comment by tsterbak** [comment]
Hey @BjornW great work!
Looks good so far, only minor changes and this is good to go :)

**Comment by tsterbak** [comment]
@BjornW any updates on this? I would love to get this merged for the next release :)

**Comment by BjornW** [comment]
> @BjornW any updates on this? I would love to get this merged for the next release :)

Hi sorry, I was sick for a few days and after that had to take of family members who also got sick. I'll have a look this evening to work on the last changes.

File: openandroidinstaller/assets/configs/sweet.yaml (hunk 3 above)
Comment (BjornW) [reply openandroidinstaller/assets/configs/sweet.yaml]: I've added the twrp-link similar to other configs, but with this device's device code.

**Comment by tsterbak** [comment]
@anon1892 Could you have a look? Then I would be happy to merge this :)

**Review by anon1892 — APPROVED** [review]
LGTM

File: openandroidinstaller/assets/configs/sweet.yaml (hunk 3 above)
Comment (tsterbak) [reply openandroidinstaller/assets/configs/sweet.yaml]: You are right :+1:

**Pull request merged**